cmd/scaffold.go

```diff
@@ -682,15 +684,32 @@ func (fnb *FlowNodeBuilder) Module(name string, f func(builder NodeBuilder, node
 	return fnb
 }
 
-// MustNot asserts that the given error must not occur.
-//
-// If the error is nil, returns a nil log event (which acts as a no-op).
-// If the error is not nil, returns a fatal log event containing the error.
-func (fnb *FlowNodeBuilder) MustNot(err error) *zerolog.Event {
+// Errors returns a channel that receives any unrecoverable errors encountered by the builder
+func (fnb *FlowNodeBuilder) Errors() <-chan error {
+	return fnb.errorManager.Errors()
+}
+
+// handle an unrecoverable error. if an error manager is configured, pass error to the manager, otherwise log a panic
+func (fnb *FlowNodeBuilder) ThrowError(err error, msg string) {
+	if fnb.errorManager == nil || !fnb.lm.StartupCommenced() {
```

Comment:
Let's avoid checking `lm.StartupCommenced()`, this is a field that is meant to be private, and I also don't really think this is a reliable way to determine whether we should panic or insert into the error channel.

I think we should keep `ThrowError()` as a method that is only called during initialization, which always results in a panic.

If asynchronous code needs to throw a fatal error, it can just call `fnb.errorManager.ThrowError()`, there's not really a need to wrap that with a helper function. Trying to combine the synchronous and asynchronous error throwing code into a single function obfuscates things imo

Reply:
I'm not happy with using the lm variable here either, but I also think requiring developers to know/remember to use one method vs another based on a potentially non-obvious code path will be prone to bugs.

Ultimately the goal is to refactor the initialization logic to return any errors encountered directly so the synchronous case should go away.

I'm not sure if it addresses your concerns, but we could add a `started` variable to the builder struct that's set in the 
`OnStart` closure. That at least avoids exposing the private variable from lifecycle.

Reply:
> I'm not sure if it addresses your concerns, but we could add a started variable to the builder struct that's set in the OnStart closure. That at least avoids exposing the private variable from lifecycle.

This was what I was actually initially gonna suggest as a workaround.

However, ultimately I still hold the opinion that we should just separate the two methods. As you noted, if we do separate the two methods then eventually `ThrowError()` will no longer exist anyways, because any place where it would have been called will be replaced with the error being returned directly instead. Then, only the calls to `fnb.errorManager.ThrowError()` will remain.